croncat/src/streams/tasks.rs

```diff
@@ -50,36 +58,54 @@ pub async fn tasks_loop(
     Ok(())
 }
 
-pub async fn do_task_if_any(
+pub async fn rules_loop(
     mut block_stream_rx: BlockStreamRx,
     mut shutdown_rx: ShutdownRx,
     signer: GrpcSigner,
+    block_status: Arc<Mutex<AgentStatus>>,
 ) -> Result<(), Report> {
     let block_consumer_stream = tokio::task::spawn(async move {
-        while (block_stream_rx.recv().await).is_ok() {
-            let account_addr = signer.account_id().as_ref();
-            let agent_active = signer
-                .get_agent(account_addr)
-                .await
-                .unwrap()
-                .map_or(false, |ag| ag.status == AgentStatus::Active);
-            if agent_active {
-                let tasks = signer.get_agent_tasks(account_addr).await.unwrap();
-                if tasks.is_some() {
-                    if let Ok(proxy_call_res) = signer.proxy_call().await {
-                        info!("Finished task: {}", proxy_call_res.log);
-                    } else {
-                        warn!("Something went wrong during proxy_call");
+        while let Ok(block) = block_stream_rx.recv().await {
+            let tasks_with_rules = signer.fetch_rules().await.expect("Failed to fetch rules");
```

Comment:
Don't like fetching rules every block. But there is a chance that someone completed the task, not sure what's the best thing to do here, but good enough for MVP IMO